lib/BackgroundJob/Tasks/CreateClustersTask.php

```diff
@@ -104,87 +104,53 @@ public function execute(FaceRecognitionContext $context) {
-				$oldestFaceTimestamp = $face->creationTime->getTimestamp();
-				$currentTimestamp = (new \DateTime())->getTimestamp();
-				$this->logDebug(sprintf('Oldest face without persons for user %s and model %d is from %s',
-					$userId, $modelId, $face->creationTime->format('Y-m-d H:i:s')));
-				// todo: get rid of magic numbers (move to config)
-				if ($currentTimestamp - $oldestFaceTimestamp > 2 * 60 * 60) {
-					$haveNewFaces = true;
-				}
-			}
-
-			$stalePersonsCount = $this->personMapper->countPersons($userId, $modelId, true);
-			$haveStalePersons = $stalePersonsCount > 0;
-			$staleCluster = $haveStalePersons === false && $haveNewFaces === false;
-
-			$forceRecreation = $this->settingsService->getNeedRecreateClusters($userId);
+			$forceRecreate = $this->needRecreateBySettings($userId);
+			$haveEnoughFaces = $this->hasNewFacesToRecreate($userId, $modelId);
+			$haveStaled = $this->hasStalePersonsToRecreate($userId, $modelId);
 
-			$this->logDebug(sprintf('Found %d changed persons for user %s and model %d', $stalePersonsCount, $userId, $modelId));
-
-			if ($staleCluster && !$forceRecreation) {
+			if ($forceRecreate) {
+				$this->logInfo('Clusters already exist, but there was some change that requires recreating the clusters');
+			}
+			else if (!$haveStaled && !$haveEnoughFaces) {
 				// If there is no invalid persons, and there is no recent new faces, no need to recreate cluster
 				$this->logInfo('Clusters already exist, estimated there is no need to recreate them');
 				return;
 			}
-			else if ($forceRecreation) {
-				$this->logInfo('Clusters already exist, but there was some change that requires recreating the clusters');
+			else {
+				$this->logInfo('Face clustering will be recreated.');
 			}
-		} else {
+		}
+		else {
 			// User should not be able to use this directly, used in tests
-			$forceCreation = $this->settingsService->getForceCreateClusters($userId);
-
-			// These are basic criteria without which we should not even consider creating clusters.
-			// These clusters will be small and not "stable" enough and we should better wait for more images to come.
-			// todo: 2 queries to get these 2 counts, can we do this smarter?
-			$imageCount = $this->imageMapper->countUserImages($userId, $modelId);
-			$imageProcessed = $this->imageMapper->countUserProcessedImages($userId, $modelId);
-			$percentImagesProcessed = 0;
-			if ($imageCount > 0) {
-				$percentImagesProcessed = $imageProcessed / floatval($imageCount);
+			$forceTestCreation = $this->settingsService->_getForceCreateClusters($userId);
+			$needCreate = $this->needCreateFirstTime($userId, $modelId);
+
+			if ($forceTestCreation) {
+				$this->logInfo('Force the creation of clusters for testing');
 			}
-			$facesCount = $this->faceMapper->countFaces($userId, $modelId);
-			// todo: get rid of magic numbers (move to config)
-			if (!$forceCreation && ($facesCount < 1000) && ($imageCount < 100) && ($percentImagesProcessed < 0.95)) {
```

Comment:
@stalker314314 
Do you remember what your original intention here?
As it was (and still in this PR), having 100 photos, analyzed or not, always creates the initial clusters.